### ISSUE TITLE
🐛 Panic `index out of bounds: the len is 4 but the index is 4` in `crates/biome_control_flow/src/lib.rs`

### ISSUE THREAD

**Issue by qarmin** [biomejs/biome#6640]
### Environment information

```block
Self compiled c0718ca610a655e675182ac6c0424301aa64c325
```

### What happened?

File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```
 if)for(s of=>{  for(t of break
```

command
```
timeout -v 150 biome lint TEST___FILE.js
```

cause this
```
Checked 0 files in 37ms. No fixes applied.

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /home/runner/work/Automated-Fuzzer/Automated-Fuzzer/biome-main/crates/biome_control_flow/src/lib.rs:36:21
Thread Name: biome::workspace_worker_1
Message: index out of bounds: the len is 4 but the index is 4

/opt/BROKEN_FILES_DIR/matchName_IDX_0_RAND_52369521854534532731688_minimized_62.js internalError/panic  INTERNAL  ━━━━━━━━━━

  × processing panicked: index out of bounds: the len is 4 but the index is 4
  
  ! This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × No files were processed in the specified paths.
  
  i Check your biome.json or biome.jsonc to ensure the paths are not ignored by the configuration.
  
  i These paths were provided but ignored:
  
  - /opt/BROKEN_FILES_DIR/matchName_IDX_0_RAND_52369521854534532731688_minimized_62.js
  


##### Automatic Fuzzer note, output status "Some(1)", output signal "None"
```

[compressed.zip](https://github.com/user-attachments/files/20977429/compressed.zip)

### Expected result

No panic

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
